### PR TITLE
fix: Configure Amplify categories

### PR DESCRIPTION
**Pull request by nadetastic**
#### Related GitHub issue #, if available:
https://github.com/aws-amplify/docs/issues/6570
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/tools/libraries/configure-categories/index.mdx [hunk 1]
```diff
@@ -26,10 +19,207 @@ export function getStaticProps(context) {
   };
 }
 
-import js0 from '/src/fragments/lib-v1/client-configuration/js/js-configuration.mdx'; 
+<InlineFilter filters={['javascript', 'react-native', 'angular', 'nextjs', 'vue','react']}>
 
-<Fragments fragments={{ javascript: js0, angular: js0, nextjs: js0, react: js0, vue: js0 }} />
+When using the Amplify CLI, the **aws-exports.js** file gets created and updated automatically for you based upon the resources you have added and configured.
```
Comment (israx) [inline src/pages/[platform]/tools/libraries/configure-categories/index.mdx:24]: I might be wrong but I think the `aws-exports.js` got renamed to `amplifyConfiguration`.

File: src/pages/[platform]/tools/libraries/configure-categories/index.mdx [hunk 2]
```diff
@@ -26,10 +19,207 @@ export function getStaticProps(context) {
...
-<Fragments fragments={{ 'react-native': reactnative0 }} />
\ No newline at end of file
+## General configuration
+
+The `Amplify.configure()` configuration can generally be included in your project _entrypoint_ file (i.e. _/src/main.js_). It takes in two parameters - `ResourcesConfig` and `LibraryOptions`.
+
+```ts
+Amplify.configure(
+  {}: ResourcesConfig;
+  {}?: LibraryOptions;
```
Comment (israx) [inline src/pages/[platform]/tools/libraries/configure-categories/index.mdx:35]: I believe we don't export `LibraryOptions` yet

File: src/pages/[platform]/tools/libraries/configure-categories/index.mdx [hunk 3]
```diff
@@ -26,10 +19,207 @@ export function getStaticProps(context) {
...
+
+## Interactions
+
+### Top level configuration
+
+```js
+Amplify.configure({
+  Auth: {
+    identityPoolId: 'us-east-1:xxx-xxx-xxx-xxx-xxx', // (required) Identity Pool ID
+    region: 'us-east-1' // (required) Identity Pool region
```
Comment (israx) [inline src/pages/[platform]/tools/libraries/configure-categories/index.mdx:211]: I don't think the config accepts a region

File: src/pages/[platform]/tools/libraries/configure-categories/index.mdx [hunk 4]
```diff
@@ -26,10 +19,207 @@ export function getStaticProps(context) {
...
+      signUpVerificationMethod?: 'code' | 'link';
+      loginWith?: {
+        oauth?: OAuthConfig;
+        username?: boolean;
+        email?: boolean;
+        phone?: boolean;
+      };
+    }
+  }
+});
```
Comment (israx) [inline src/pages/[platform]/tools/libraries/configure-categories/index.mdx:57]: This is defining the types but not an example.

File: src/pages/[platform]/tools/libraries/configure-categories/index.mdx [hunk 5]
```diff
@@ -26,10 +19,207 @@ export function getStaticProps(context) {
...
+      interpretText: {
+        region: 'us-east-1',
+        proxy: false,
+        defaults: {
+          type: 'ALL'
+        }
+      }
+    }
+  }
+});
```
Comment (israx) [inline src/pages/[platform]/tools/libraries/configure-categories/index.mdx:172]: I think this is how the other configs should be defined too. Here we are actually creating an example

**Review by israx — COMMENTED** [review]
I think we should actually create examples and not provide only the type definitions for some categories. The current changes are inconsistent between categories, for instance `Auth` provides an example with type definitions whereas `Predictions` defines the actual config values.

File: src/pages/[platform]/tools/libraries/configure-categories/index.mdx (hunk 1 above)
Comment (nadetastic) [reply src/pages/[platform]/tools/libraries/configure-categories/index.mdx:24]: Good catch ty

File: src/pages/[platform]/tools/libraries/configure-categories/index.mdx (hunk 4 above)
Comment (nadetastic) [reply src/pages/[platform]/tools/libraries/configure-categories/index.mdx:57]: Good point, will update to show example values

File: src/pages/[platform]/prev/tools/libraries/configure-categories/index.mdx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Configure Amplify categories',
+  description: 'Configuring the client.',
+  platforms: ['javascript', 'react-native', 'angular', 'nextjs', 'vue']
```
Comment (renebrandel) [inline src/pages/[platform]/prev/tools/libraries/configure-categories/index.mdx:6]: react is missing here

**Review by renebrandel — APPROVED** [review]

File: src/pages/[platform]/tools/libraries/configure-categories/index.mdx [hunk 7]
```diff
@@ -26,10 +19,262 @@ export function getStaticProps(context) {
...
+## Geo
+
+```js
+Amplify.configure({
+  Geo: {
+    LocationService: {
+      region: 'us-east-1',
+      maps: {
+        // Optional
+        items: {},
```
Comment (kvramyasri7) [inline src/pages/[platform]/tools/libraries/configure-categories/index.mdx:262]: suggestion: good see some data in these fields.

**Review by kvramyasri7 — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**